### PR TITLE
Beliefsdataframe initialisation improvements

### PR DESCRIPTION
**Pull request by Flix6x**

### CODE REVIEW

File: timely_beliefs/beliefs/utils.py [hunk 1]
```diff
@@ -313,20 +312,33 @@ def resample_event_start(
...
+    belief_horizon: Union[timedelta, pd.Series],
     cumulative_probability: float = 0.5,
 ) -> List["classes.TimedBelief"]:
     """Turn series entries into TimedBelief objects."""
     beliefs = []
-    for time, value in event_value_series.items():
+    if isinstance(belief_horizon, timedelta):
+        belief_horizon_series = pd.Series(belief_horizon)
+    else:
+        belief_horizon_series = belief_horizon
```
Comment (nhoening) [inline timely_beliefs/beliefs/utils.py:324]: I guess we should check if `belief_horizon_series` end up with the same number of elements as `event_value_series`.

File: timely_beliefs/beliefs/utils.py [hunk 2]
```diff
@@ -313,20 +312,33 @@ def resample_event_start(
...
+    if isinstance(source, BeliefSource):
+        source_series = pd.Series(BeliefSource)
+    else:
+        source_series = source
```
Comment (nhoening) [inline timely_beliefs/beliefs/utils.py:328]: I guess we should check if `source_series` end up with the same number of elements as `event_value_series`.

File: timely_beliefs/beliefs/utils.py (hunk 1 above)
Comment (nhoening) [reply timely_beliefs/beliefs/utils.py:324]: If it's only a timedelta, shouldn't we repeat the value n times (n=-length of event_value_series)?

File: timely_beliefs/utils.py [hunk 3]
```diff
@@ -5,12 +5,12 @@
 import pandas as pd
 
 
-def enforce_utc(dt: datetime) -> pd.Timestamp:
+def enforce_tz(dt: datetime, name: str) -> pd.Timestamp:
```
[the remark this answers is not in the document]
Comment (Flix6x) [reply timely_beliefs/utils.py:8]: Indeed, this function now serves only to verify a timezone is set.

It seems changing types isn't actually needed anymore. I'm not sure why I made a pandas Timestamp in the first place; maybe it was needed with an earlier version of pandas (I believe I've seen improved support for timezone aware datetimes).

**Pull request merged**